In [1]:
import pandas as pd
import numpy as np
from itables import show

In [7]:
df = pd.read_parquet('../data/spend_data_large_sourcing.parquet')
show(df)
df.shape

(698411, 51)

## Remove Unneeded Columns
### Needed Columns
1. Spend_Data_Posting_Date_Fiscal_Year
2. Spend_Data_Posting_Date_Fiscal_Year_and_Quarter
3. Spend_Data_Posting_Date
4. Spend_Data_Vendor_Invoice_Date
5. Spend_Data_PO_Number
6. Spend_Data_PO_Line_Item_Number
7. Spend_Data_Vendor_Number
8. Spend_Data_Invoice_Type
9. Spend_Data_Business_Partner_Name_
10. Spend_Data_Organization
11. Spend_Data_Division_Description
12. Spend_Data_Business_Unit_Description
13. Spend_Data_Geo
14. Spend_Data_Line_Item_Text
15. Spend_Data_Vendor_Invoice_Amount_LC2_USD_
16. Spend_Data_Vendor_Name
17. Spend_Data_Sourcing_Category
18. Spend_Data_Sourcing_Portfolio

## Filter Down Spend Data - Starting Point (698,411 rows; 51 columns)
1. Remove rows where Spend_Data_Invoice_Type is 'NonPO-Related'
2. Spend_Data_Sourcing_Portfolio = 'Technology


In [15]:
#Options available in Spend_Data_Sourcing_Category
df['Spend_Data_Sourcing_Category'].unique()

array(['Enterprise managed software', 'Network and telecom',
       'Technology professional services', 'Business process outsourcing',
       'Infrastructure', 'Hardware', 'Reseller', 'Royalties'],
      dtype=object)

In [16]:
#1 remove rows where Spend_Data_Invoice_Type is 'NonPO-Related'
df = df[df['Spend_Data_Invoice_Type'] != 'NonPO-Related']
print(df.shape)

#2 remove rows where Spend_Data_Sourcing_Portfolio is not 'Technology'
df = df[df['Spend_Data_Sourcing_Portfolio'] == 'Technology']
print(df.shape)

#3 filter down to Enterprise sourcing category
df = df[df['Spend_Data_Sourcing_Category'] == 'Enterprise managed software']
print(df.shape)


(76136, 51)
(76136, 51)
(8082, 51)


In [18]:
#Count unique vaules with (Vendor_Name, Spend_Data_Sourcing_Category)
print("Number of unique values with (Vendor_Name, Spend_Data_Sourcing_Category, line item text):")
df.groupby(['Spend_Data_Vendor_Name', 'Spend_Data_Sourcing_Category', 'Spend_Data_Line_Item_Text']).ngroups


Number of unique values with (Vendor_Name, Spend_Data_Sourcing_Category, line item text):


4287